lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorManager.java

```diff
@@ -460,9 +460,18 @@ private void onTaskStatusMessage(final ReefServiceProtos.TaskStatusProto taskSta
     if (!(this.task.isPresent() && this.task.get().getId().equals(taskStatusProto.getTaskId()))) {
       if (taskStatusProto.getState() == ReefServiceProtos.State.INIT ||
           taskStatusProto.getState() == ReefServiceProtos.State.FAILED ||
+          taskStatusProto.getState() == ReefServiceProtos.State.RUNNING ||
           taskStatusProto.getRecovery() // for task from recovered evaluators
           ) {
 
+        // REEF-308 exposes a bug where the .NET evaluator does not send its states in the right order
```

Comment:
Nitpick: Use `[REEF-308]` such that we can find these lines later when we fix the root cause. Also, add a reference to `[REEF-289]` which is the likely root cause here.